src/edu/washington/escience/myria/operator/CSVFragmentTupleSource.java

```diff
@@ -409,24 +409,28 @@ protected void init(final ImmutableMap<String, Object> execEnvVars) throws DbExc
       int workerID = getNodeID();
       long fileSize = source.getFileSize();
       long currentPartitionSize = fileSize / workerIds.length;
-      int workerIndex = 0;
+      int workerIndex = -1;
       for (int i = 0; i < workerIds.length; i++) {
         if (workerID == workerIds[i]) {
           workerIndex = i;
         }
       }
-      boolean isLastWorker = workerIndex == workerIds.length - 1;
-      long startByteRange = currentPartitionSize * workerIndex;
-      long endByteRange;
+      if (workerIndex >= 0) {
+        boolean isLastWorker = workerIndex == workerIds.length - 1;
```

Comment:
Parens around the condition would be more readable (I realize this was in the original).